MycroftAI/mycroft-core
# Feature/mark 2

## Description
Adds a generic hardware enclosure capability for input (keyboards, mice) and output (screens) devices

## How to test
If working you should see output like this in your log file 

==> /var/log/mycroft/enclosure.log <==
2021-02-09 21:11:50.585 | INFO     | 26731 | mycroft.enclosure.hardware_enclosure:__init__:85 | Mark2 HardwareEnclosure() initialized
2021-02-09 21:11:50.660 | INFO     | 26731 | mycroft.client.enclosure.mark2.interface:__init__:60 | ** EnclosureMark2 initalized **
2021-02-09 21:11:50.804 | INFO     | 26731 | __main__:main:84 | Enclosure created, capabilities ===>{'keyboards': [], 'mice': [{'name': 'FT5406 memory based driver', 'extra': 'Touch Screen'}], 'screens': [{'name': 'DRM emulated', 'resolution': '800,480', 'pel_size': '32', 'extra': ''}]}
2021-02-09 21:11:50.807 | INFO     | 26731 | __main__:main:87 | Mark2 detected[sj201r4], additional capabilities ===>{'Led': {'name': 'led_sj201r4', 'type': 'MycroftLed'}, 'Switch': {'name': 'switch_gpio', 'type': 'MycroftSwitch'}, 'Volume': {'name': 'volume_sj201r4', 'type': 'MycroftVolume'}, 'Palette': {'name': 'default_palette', 'type': 'MycroftPalette'}}
2021-02-09 21:11:50.811 | INFO     | 26731 | __main__:main:88 | Leds ===>{'num_leds': 10, 'brightness': '(0.0-1.0)', 'led_colors': 'MycroftPalette', 'reserved_leds': [10, 11]}
2021-02-09 21:11:50.818 | INFO     | 26731 | __main__:main:89 | Volume ===>{'range': (0.0, 1.0), 'type': 'MycroftTIAmp'}
2021-02-09 21:11:50.819 | INFO     | 26731 | __main__:main:90 | Switches ===>{'user_volup_handler': 'button', 'user_voldown_handler': 'button', 'user_action_handler': 'button', 'user_mute_handler': 'slider'}
2021-02-09 21:11:50.821 | INFO     | 26731 | __main__:main:93 | __main__().py Starting Client Enclosure!
2021-02-09 21:11:50.839 | INFO     | 26731 | mycroft.messagebus.client.client:on_open:114 | Connected
2021-02-09 21:11:50.847 | INFO     | 26731 | mycroft.services.util:start_message_bus_client:43 | Connected to messagebus
2021-02-09 21:11:50.848 | INFO     | 26731 | __main__:on_ready:26 | Enclosure started!


## Contributor license agreement signed?
Yes
